pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -230,14 +232,64 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 		return configMetaErr
 	}
 
-	if state == clusterMembersChangedUnready {
+	if state == clusterMembersChangedUnready || (cr.Spec.ConfigMetaGenerator != cr.Status.LastConfigMetaGenerator) {
 		cr.Status.SpecGenerationToProcess = &cr.Generation
 	}
 	membersErr := syncMembers(reqLogger, cr, roles, configmetaGen)
 	if membersErr != nil {
 		errLog("members", membersErr)
 		return membersErr
 	}
+	cr.Status.LastConfigMetaGenerator = cr.Spec.ConfigMetaGenerator
+	cr.Status.State = string(clusterReady)
```

Comment:
We cannot set clusterReady status here. We may not be done processing changes on a single handler pass.